lib/ui/process_ui/demographic_details/age_date.dart

```diff
@@ -0,0 +1,132 @@
+import 'package:flutter/material.dart';
+
+class AgeDate extends StatefulWidget {
+  const AgeDate({super.key, required this.validation, required this.onChanged});
+
+  final RegExp validation;
+  final Function(String) onChanged;
+
+  @override
+  State<AgeDate> createState() => _AgeDateState();
+}
+
+class _AgeDateState extends State<AgeDate> {
+  final TextEditingController _dayController = TextEditingController();
+
+  final TextEditingController _monthController = TextEditingController();
+
+  final TextEditingController _yearController = TextEditingController();
+
+  final TextEditingController _ageController = TextEditingController();
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        Flexible(
+          child: Container(
+            padding: const EdgeInsets.symmetric(horizontal: 16.0, vertical: 0),
+            decoration: BoxDecoration(
+              border: Border.all(color: const Color(0xff9B9B9F), width: 1),
+              borderRadius: BorderRadius.circular(8.0),
+            ),
+            child: TextFormField(
+              validator: (value) {
+                if (value == null || value.isEmpty) {
+                  return 'Please enter a value';
+                }
+                if (!widget.validation.hasMatch(
+                    "${_yearController.text}/${_monthController.text}/${_dayController.text}")) {
+                  return 'DD';
```

Comment:
hints provided should be in registration language.

Reply:
currently working on this